engine/clients/redis/parser.py

```diff
@@ -0,0 +1,85 @@
+import json
+from collections import ChainMap
+from typing import Any, Dict, List, Optional, Text, Tuple
+
+from engine.base_client.parser import BaseConditionParser, FieldValue
+
+QueryParamsTuple = Tuple[Text, Dict[Text, Any]]
+
+
+class RedisConditionParser(BaseConditionParser):
+    def __init__(self) -> None:
+        super().__init__()
+        self.counter = 0
+
+    def build_condition(
+        self,
+        and_statements: List[QueryParamsTuple],
+        or_statements: List[QueryParamsTuple],
+    ) -> Tuple[Text, Dict[Text, Any]]:
+        if and_statements is None or 0 == len(and_statements):
```

Comment:
type hints do not imply None as a possible value

these conditions are bulky, maybe we can replace them with just `if not and_statements`?

Reply:
maybe even replace this code with just `and_clauses, and_params = list(zip(*and_statements)) if and_statements else ([], [])`